libs/core/src/lib/fixed-card-layout/fixed-card-layout.component.ts

```diff
@@ -0,0 +1,191 @@
+import { CdkDragDrop, CdkDropList, moveItemInArray } from '@angular/cdk/drag-drop';
+import {
+    AfterContentInit,
+    AfterViewChecked,
+    AfterViewInit,
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ContentChildren,
+    Directive,
+    ElementRef,
+    EventEmitter,
+    Input,
+    OnDestroy,
+    OnInit,
+    Output,
+    QueryList,
+    TemplateRef,
+    ViewChild
+} from '@angular/core';
+import { Observable, of, from, fromEvent, Subscription, Subject } from 'rxjs';
+import { debounceTime, takeUntil } from 'rxjs/operators';
+
+const CARD_MINIMUM_WIDTH = 320; // in px; 20rem max card size
+const CARD_GAP_WIDTH = 16; // gap=1rem==16px
+
+@Directive({ selector: '[fdCardDef]' })
+export class CardDefinitionDirective {
+    constructor(public template: TemplateRef<any>) {}
+}
+
+class Layout {
+    constructor(public noOfColumns: number, public screenSize: number) {}
+}
+
+class CardDropped {
+    constructor(public container: CdkDropList, public previousIndex: number, public currentIndex: number) {}
+}
+
+type CardColumn = CardDefinitionDirective[];
+
+@Component({
+    selector: 'fd-fixed-card-layout',
+    templateUrl: './fixed-card-layout.component.html',
+    styleUrls: ['./fixed-card-layout.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class FixedCardLayoutComponent implements OnInit, AfterContentInit, AfterViewInit, AfterViewChecked, OnDestroy {
+    /** @hidden */
+    @ContentChildren(CardDefinitionDirective)
+    cards: QueryList<CardDefinitionDirective>;
+
+    /** @hidden */
+    @ViewChild('layout')
+    layout: ElementRef;
+
+    /** Drag drop behaviour can be disabled */
+    @Input()
+    disableDragDrop: boolean;
+
+    /** Event to emit, when layout changes */
+    @Output()
+    layoutChange: EventEmitter<Layout> = new EventEmitter<Layout>();
+
+    /** Event to emit on Card dragged and dropped */
+    @Output()
+    cardDraggedDropped: EventEmitter<CardDropped> = new EventEmitter<CardDropped>();
+
+    /** Array of CardDefinitionDirective Array.To make Table kind of layout.*/
+    public columns: CardColumn[];
+    /** @hidden Number of Columns in layout */
+    private numberOfColumns = 3;
+    /** @hidden */
+    private previousNumberOfColumns: number;
+
+    /** An RxJS Subject that will kill the data stream upon component’s destruction (for unsubscribing)  */
+    private readonly _onDestroy$: Subject<void> = new Subject<void>();
+
+    constructor(private readonly _cd: ChangeDetectorRef) {}
+
+    ngOnInit(): void {
+        fromEvent(window, 'resize')
+            .pipe(debounceTime(60), takeUntil(this._onDestroy$))
+            .subscribe(() => this.onResize());
+    }
+
+    /** @hidden */
+    ngAfterContentInit(): void {
+        // subscribe to number of card change
+        this.cards.changes.subscribe(() => this._renderLayout());
```

Comment:
This is not a big issue, but lets assume your component is in following state:
```
[ card ] [ card ] [ card ] [ card ]
```
And then number of cards changes to 1:
```
[ card ] [  ] [  ] [  ]
```
Leaving you with empty containers for columns because you don't re-calculate number of containers on cards change.

Reply:
currently when hiding cards, width of layout does not change. so number of columns returned remains same.
Mostly we will not such usecase from current requirement but
I will raise an issue for tracking this, i will fix in next phase.

Reply:
https://github.com/SAP/fundamental-ngx/issues/3437 issue raised for tracking this.